### PR TITLE
feat: [DHIS2-18081] Problem with date format when scheduling an event

### PR DESCRIPTION
**Pull request by henrikmv**
[DHIS2-18081](https://dhis2.atlassian.net/browse/DHIS2-18081)

- Converts date to server format `YYYY-MM-DD`.
- Format of date stored in the scheduleDate-state is based on the system settings.

[DHIS2-18081]: https://dhis2.atlassian.net/browse/DHIS2-18081?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/utils/converters/date/stringToMomentDateFormat.js [hunk 1]
```diff
@@ -11,6 +11,6 @@ import { systemSettingsStore } from '../../../metaDataMemoryStores';
 export function convertStringToDateFormat(date: string) {
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/utils/converters/date/stringToMomentDateFormat.js:11]: With this change, I think a more suitable name for this function is `convertStringToServerFormat`. The function's comment definition also needs to be updated. WDYT?

File: src/core_modules/capture-core/components/WidgetEventSchedule/WidgetEventSchedule.container.js [hunk 2]
```diff
@@ -45,13 +46,15 @@ export const WidgetEventSchedule = ({
     });
     const { currentUser, noteId } = useNoteDetails();
     const [scheduleDate, setScheduleDate] = useState('');
+    const dateFormat = systemSettingsStore.get().dateFormat;
+    const formatDateForServer = date => moment(date, dateFormat).format('YYYY-MM-DD');
+    const formattedScheduleDate = formatDateForServer(scheduleDate);
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetEventSchedule/WidgetEventSchedule.container.js:51]: Hi @henrikmv 
Can you use existing converters instead, in the same way as is currently done elsewhere, i.e. [getConvertedRelatedStageEvent](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/getConvertedRelatedStageEvent/getConvertedRelatedStageEvent.js#L54)?

```
const convertFn = pipe(convertFormToClient, convertClientToServer);
const formattedScheduleDate = convertFn(scheduleDate, dataElementTypes.DATE);
```

Thanks!

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Review by eirikhaugstulen — APPROVED** [review]
Looking good, Henrik!

**Pull request merged**